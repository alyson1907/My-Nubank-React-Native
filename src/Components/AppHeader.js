import React from 'react'
import { View, Image } from 'react-native'
import Text from './Text'
// Images/Assets
import logo from '../img/logo/Nubank_Logo.png'
import Icon from 'react-native-vector-icons/MaterialIcons'
import styles from './Styles/AppHeader'
import Colors from '../Colors'

const AppHeader = () => {
  return (
    <View style={styles.container}>
      <View style={styles.innerView}>
        <Image source={logo} />
        <Text style={styles.textName}>Alyson</Text>
      </View>
      <Icon name="keyboard-arrow-down" size={24} color={Colors.secondary} />
    </View>
  )
}

export default AppHeader
