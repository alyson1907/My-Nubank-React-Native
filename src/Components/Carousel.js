import React, { useState, useEffect } from 'react'
import { Animated, View, Text, ART, Dimensions } from 'react-native'
import { ScrollView } from 'react-native-gesture-handler'
import styles from './Styles/Carousel'

const Carousel = (props) => {
  const hasChildren = !!props.children
  const numberOfItems = hasChildren && props.children.length || 0

  const [selectedBullet, setSelectedBullet] = useState(0)
  const [width, setWidth] = useState(Dimensions.get('window').width)

  const handleBullet = e => {
    const getItemIndex = e => {
      const contentWidth = width
      // const { width: contentWidth } = e.nativeEvent.contentSize
      const { x: contentOffset } = e.nativeEvent.contentOffset

      const pageWidth = parseInt(contentWidth / numberOfItems)
      const currentPage = Math.round(contentOffset / pageWidth)

      return currentPage
    }
    const currentBullet = getItemIndex(e)
    // TODO add Animation to bullet change
    setSelectedBullet(currentBullet)
  }

  const getPageBullets = (numberOfItems, selectedBullet) => {
    return (
      <View style={styles.bulletsContainer}>
        {hasChildren && props.children.map((child, idx) => {
          return idx === selectedBullet ? (<View key={idx} style={styles.selectedBullet} />) : (<View key={idx} style={styles.bullet} />)
        })}
      </View>
    )
  }

  return (
    <View style={props.style}>
      <ScrollView horizontal={true}
        showsHorizontalScrollIndicator={false}
        contentContainerStyle={{ width: `${100 * numberOfItems}%`, justifyContent: 'space-around' }}
        onContentSizeChange={(width, height) => setWidth(width)}
        decelerationRate="normal"
        pagingEnabled
        onScroll={handleBullet}
      >
        {props.children}
      </ScrollView>
      {props.bullets && getPageBullets(numberOfItems, selectedBullet)}
    </View>
  )
}

export default Carousel
