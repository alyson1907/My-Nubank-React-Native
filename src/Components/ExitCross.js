import React from 'react'
import { TouchableOpacity } from 'react-native-gesture-handler'
import { useNavigation } from '@react-navigation/native'
import Icon from 'react-native-vector-icons/MaterialIcons'
import Colors from '../Colors'

export default (props) => {
  const navigation = useNavigation()
  const returnScreen = () => navigation.canGoBack() && navigation.goBack()

  return (
    <TouchableOpacity onPress={returnScreen}>
      <Icon name="close" size={32} color={Colors.fontSecondary} />
    </TouchableOpacity>
  )
}
