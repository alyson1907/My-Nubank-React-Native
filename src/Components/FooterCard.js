import React from 'react'
import Text from './Text'
import { Image } from 'react-native'
import { useNavigation } from '@react-navigation/native'
import { TouchableOpacity } from 'react-native-gesture-handler'

import Icon from 'react-native-vector-icons/AntDesign'
import Colors from '../Colors'
import styles from './Styles/FooterCard'

const FooterCard = (props) => {
  const navigation = useNavigation()
  const navigateTo = pageName => pageName && navigation.navigate(pageName)

  return (
    // TODO add link to navigate to another screen
    <TouchableOpacity style={styles.container} onPress={() => navigateTo(props.navigateTo)}>
      <Icon name={props.icon} size={22} color={Colors.secondary} />
      <Text style={styles.contentText}>{props.text}</Text>
    </TouchableOpacity>
  )
}

export default FooterCard
