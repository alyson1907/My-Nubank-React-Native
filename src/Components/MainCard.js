import React from 'react'
import { View, Image } from 'react-native'
import Text from './Text'
import { useNavigation } from '@react-navigation/native'

import Icon from 'react-native-vector-icons/FontAwesome5'
import Colors from '../Colors'
import styles from './Styles/MainCard'

const MainCard = (props) => {
  const navigation = useNavigation()
  const navigateTo = pageName => pageName && navigation.navigate(pageName)

  const header = props => (
    <View style={styles.headerContainer}>
      {props.header && props.header.icon && (<Icon name={props.header.icon} size={28} color={Colors.fontSecondary} />)}
      <Text style={styles.headerText}>{props.header.text}</Text>
    </View>
  )

  const footer = props => {
    if (props.footer) {
      return (
        <View style={styles.footer}>
          <Icon name={props.footer.icon} size={28} color={Colors.fontSecondary} />
          <Text style={styles.footerTxt}>{props.footer.text}</Text>
          <Icon name='arrow-right' size={18} color={Colors.fontSecondary} />
        </View>
      )
    }
  }

  return (
    <View style={styles.container} onTouchEnd={() => {navigateTo(props.navigateTo)}}>
      {/* Card Header */}
      {header(props)}

      {/* Card Content */}
      {props.children}

      {/* Card Footer */}
      {footer(props)}
    </View>
  )
}

export default MainCard
