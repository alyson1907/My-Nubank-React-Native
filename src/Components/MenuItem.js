import React from 'react'
import Text from './Text'
import { View } from 'react-native'
import { TouchableOpacity } from 'react-native-gesture-handler'
import { useNavigation } from '@react-navigation/native'


import Icon from 'react-native-vector-icons/MaterialIcons'
import styles from './Styles/MenuItem'

const MenuItem = (props) => {
  const navigation = useNavigation()
  const navigateTo = pageName => props.navigateTo && navigation.navigate(pageName)

  return (
    <View style={styles.item}>
      <TouchableOpacity style={styles.button} onPress={() => navigateTo(props.navigateTo)}>
        <Icon name={props.icon} style={styles.icon} size={22} />
        <Text style={styles.text}>{props.text}</Text>
      </TouchableOpacity>
    </View>
  )
}

export default MenuItem
