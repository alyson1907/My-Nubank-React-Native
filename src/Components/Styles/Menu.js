import { StyleSheet } from 'react-native'
import Colors from '../../Colors'

const styles = StyleSheet.create({
  container: {
    maxHeight: 0
  }
})

export default styles
