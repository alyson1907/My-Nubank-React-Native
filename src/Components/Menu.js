import React from 'react'
import { View } from 'react-native'
import { TouchableOpacity } from 'react-native-gesture-handler'
// Components
import MenuItem from './MenuItem'

import Icon from 'react-native-vector-icons/MaterialIcons'
import styles from './Styles/Menu'

const Menu = () => {
  return (
    <View style={styles.container}>
      <MenuItem icon='help-outline' text='Me ajuda' />
      <MenuItem icon='person-outline' text='Perfil' />
      <MenuItem icon='credit-card' text='Configurar cartão' />
      <MenuItem icon='smartphone' text='Configurações do App' />
    </View>
  )
}

export default Menu
