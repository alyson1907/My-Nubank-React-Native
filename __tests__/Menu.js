import React from 'react'
import 'react-native'
import { shallow } from 'enzyme'

import Menu from '../src/Components/Menu'
// Requiring module in order to mock
import * as menuModule from '../src/Components/Menu'
import { menuTestSet } from './fixtures/menu'

const translateYMock = {
  interpolate: () => { }
}

describe('Menu Tests', () => {
  it('renders Menu and compares with snapshot', () => {
    const wrapper = shallow(<Menu translateY={translateYMock} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('asserts the onLayout handler is calls the function to update height state', () => {
    // Requiring the real implementation
    const { handleLayout } = jest.requireActual('../src/Components/Menu')

    menuTestSet.forEach(test => {
      const { input: { event } } = test
      const setHeightMock = jest.fn()
      handleLayout(event, setHeightMock)
      expect(setHeightMock).toHaveBeenCalledWith(event.nativeEvent.layout.height)
    })
  })

  it('simulates onLayout event', () => {
    const wrapper = shallow(<Menu translateY={translateYMock} />)
    menuModule.handleLayout = jest.fn()
    wrapper.find("[testID='menu-container']").simulate('layout', menuTestSet[0].input.event)
    expect(menuModule.handleLayout).toHaveBeenCalled()
  })
})
