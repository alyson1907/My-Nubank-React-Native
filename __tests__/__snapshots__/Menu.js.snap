// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Tests renders Menu and compares with snapshot 1`] = `
<AnimatedComponent
  onLayout={[Function]}
  style={
    Array [
      Object {
        "position": "absolute",
        "width": "100%",
        "zIndex": -1,
      },
      Object {
        "top": 667,
      },
      Object {
        "opacity": undefined,
      },
    ]
  }
  testID="menu-container"
>
  <ComponentWrapper>
    <MenuItem
      icon="help-outline"
      text="Me ajuda"
    />
    <MenuItem
      icon="person-outline"
      text="Perfil"
    />
    <MenuItem
      icon="credit-card"
      text="Configurar cartão"
    />
    <MenuItem
      icon="smartphone"
      text="Configurações do App"
    />
  </ComponentWrapper>
</AnimatedComponent>
`;
